File: AdvantageFramework.Core/Database/Entities/AlertGroup.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    [Table("ALERT_GROUP")]
    public partial class AlertGroup
    {
        #region Constants



        #endregion

        #region Enum

        public enum Properties
        {
            TBD
        }


        #endregion

        #region Variables



        #endregion

        #region Properties

        [Key]
        [Column("E_GROUP")]
        [StringLength(50)]
        public string EGroup { get; set; }
        [Key]
        [Column("ALERT_CAT_ID")]
        public int AlertCatId { get; set; }
        [Column("ACTIVE_FLAG")]
        public short? ActiveFlag { get; set; }

        [ForeignKey(nameof(AlertCatId))]
        [InverseProperty(nameof(AlertCategory.AlertGroups))]
        public virtual AlertCategory AlertCat { get; set; }

        #endregion

        #region Methods



        #endregion
    }
}

File: AdvantageFramework/Database/Entities/AlertGroup.vb
Namespace Database.Entities

	<Table("EMAIL_GROUP")>
	Public Class AlertGroup
		Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

		Public Enum Properties
			Code
			EmployeeCode
			IsInactive
			IncludeOnSchedule

		End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

		<Key>
		<Required>
		<MaxLength(50)>
        <Column("EMAIL_GR_CODE", Order:=0, TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=True, DisplayFormat:="")>
		Public Property Code() As String
		<Key>
		<Required>
		<MaxLength(6)>
        <Column("EMP_CODE", Order:=1, TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=True, DisplayFormat:="")>
		Public Property EmployeeCode() As String
		<Column("INACTIVE_FLAG")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", DefaultColumnType:=AdvantageFramework.BaseClasses.DefaultColumnTypes.StandardCheckBox)>
		Public Property IsInactive() As Nullable(Of Short)
		<Column("PRIMARY_EMP")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property IncludeOnSchedule() As Nullable(Of Short)


#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.Code

        End Function
        Public Overrides Function ValidateCustomProperties(ByVal PropertyName As String, ByRef IsValid As Boolean, ByRef Value As Object) As String

            'objects
            Dim ErrorText As String = ""
            Dim PropertyValue As Object = Nothing

            Select Case PropertyName

                Case AdvantageFramework.Database.Entities.AlertGroup.Properties.Code.ToString

                    If Value Is Nothing OrElse Value.Trim = "" Then

                        IsValid = False
                        ErrorText = "Please enter a alert group code."

                    End If

                Case AdvantageFramework.Database.Entities.AlertGroup.Properties.EmployeeCode.ToString

                    If Value Is Nothing OrElse Value.Trim = "" Then

                        IsValid = False
                        ErrorText = "Please enter a employee code."

                    End If

                    If IsValid Then

                        If Me.IsEntityBeingAdded() Then

                            If (From Entity In DirectCast(Me.DbContext, AdvantageFramework.Database.DbContext).AlertGroups
                                Where Entity.Code = Me.Code AndAlso
                                          Entity.EmployeeCode = Me.EmployeeCode
                                Select Entity).Any Then

                                IsValid = False
                                ErrorText = "Please enter a unique alert group\employee code combination."

                            End If

                        End If

                    End If

            End Select

            ValidateCustomProperties = ErrorText

        End Function

#End Region

	End Class

End Namespace
